libraries/plugins/rc/rc_plugin.cpp

```diff
@@ -53,6 +53,8 @@ class rc_plugin_impl
 
       void on_first_block();
       void validate_database();
+      bool before_first_block()
+      { return (_db.count< rc_account_object >() == 0); }
```

Comment:
I think this should be done without a db read. While technically O(1) there is a good amount of indirection to read a value that we can know through observation and once set does not change.

The rc plugin already listens to the post apply block notification.

I would simply track this locally as a bool. On startup perform this check and set the bool accordingly. Then, after the first block set the bool and read from it instead of the database.

Unless there is some corner case when `_db.count< rc_account_object >() == 0` could be true after the first block. In which case, `before_first_block` should be renamed to something more accurate.

Admittedly, my solution does not cover the case when the first block is forked and an alternate first block is generated. However, because that can only happen if the initminer is misconfigured, I would consider that an ops problem and not a legitimate consensus corner case that we need to cover.